### PR TITLE
fixed Aggregate-Intellect#23 (none command) and Aggregate-Intellect#38 (OpenAI error handling)

### PR DESCRIPTION
**Pull request by YujingYang666777**

### CODE REVIEW

**Review by saminegash — CHANGES_REQUESTED** [review]
Can you rebase this plase and push?

**Comment by YujingYang666777** [comment]
The conflict is solved and it should be able to merge. please let me know if here is anything I need to change

File: apps/slackbot/tools.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def get_tools(memory):
     tools = []
 
     tools.append(ContextTool(memory=memory))
+    tools.append(UserInputTool())
```
Comment (20001LastOrder) [inline apps/slackbot/tools.py:22]: Is the UserTool supposed to b added in this pr?

**Review by 20001LastOrder — CHANGES_REQUESTED** [review]
Looks good to me but is the user tool added by intention?

**Pull request closed without merging**